File: core/src/main/java/com/mcreater/amclcore/plugin/lua/LuaBaseApiLoader.java
package com.mcreater.amclcore.plugin.lua;

import com.mcreater.amclcore.plugin.lua.lib.I18NLib;
import com.mcreater.amclcore.resources.ResourceFetcher;
import org.luaj.vm2.Globals;
import org.luaj.vm2.LoadState;
import org.luaj.vm2.compiler.LuaC;
import org.luaj.vm2.lib.jse.JsePlatform;

import java.io.InputStreamReader;

public class LuaBaseApiLoader {
    private static final Globals globals = JsePlatform.standardGlobals();

    public static void init() {
        globals.load(new I18NLib());
        LoadState.install(globals);
        LuaC.install(globals);
    }

    public static void load() {
        globals.load(new InputStreamReader(ResourceFetcher.get("assets/amclcore/test.lua")), "test.lua").call();
    }
}

File: core/src/main/java/com/mcreater/amclcore/plugin/lua/LuaBaseApiLoader.kt
package com.mcreater.amclcore.plugin.lua

import com.mcreater.amclcore.plugin.lua.lib.I18NLib
import com.mcreater.amclcore.resources.ResourceFetcher
import org.luaj.vm2.LoadState
import org.luaj.vm2.compiler.LuaC
import org.luaj.vm2.lib.jse.JsePlatform
import java.io.InputStreamReader


class LuaBaseApiLoader {
    companion object {
        @JvmStatic
        private val globals = JsePlatform.standardGlobals()

        @JvmStatic
        fun init() {
            globals.load(I18NLib())
            LoadState.install(globals)
            LuaC.install(globals)
        }

        @JvmStatic
        fun load() {
            globals.load(InputStreamReader(ResourceFetcher.get("amclcore", "test.lua")), "test.lua").call()
        }
    }
}

